hera_sim/cli_utils.py

```diff
@@ -0,0 +1,136 @@
+"""
+Module containing useful helper functions and argparsers for running 
+simulations with hera_sim via the command line.
+"""
+import copy
+import os
+import numpy as np
+from .defaults import SEASON_CONFIGS
+from .simulate import Simulator
+
+def get_filing_params(config):
+    """Extract filing parameters from a configuration dictionary."""
+
+    filing_params = dict(
+        outdir=os.getcwd(),
+        outfile_name="hera_sim_simulation.uvh5",
+        output_format="uvh5",
+        clobber=False,
+    )
+    filing_params.update(config.get("filing", {}))
+    if filing_params["output_format"] not in ("miriad", "uvfits", "uvh5"):
+        raise ValueError(
+            "Output format not supported. Please use miriad, uvfits, or uvh5."
+        )
+
+    return filing_params
+
+def validate_config(config):
+    """Validate the contents of a loaded configuration file."""
+    if config.get("defaults", None) is not None:
+        if type(config["defaults"]) is not str:
+            raise ValueError(
+                "Defaults in the CLI may only be specified using a string."
+            )
+        
+        if config["defaults"] in SEASON_CONFIGS.keys():
+            return
+        else:
+            raise ValueError("Default configuration string not recognized.")
+
+    freq_params = config.get("freq", {})
+    time_params = config.get("time", {})
+    array_params = config.get("telescope", {}).get("array_layout", {})
+    if any(param == {} for param in (freq_params, time_params, array_params)):
+        raise ValueError("Insufficient information for initializing simulation.")
+
+    freqs_ok = _validate_freq_params(freq_params)
+    times_ok = _validate_time_params(time_params)
+    array_ok = _validate_array_params(array_params)
+    if not all(freqs_ok, times_ok, array_ok):
+        raise ValueError("Insufficient information to initialize simulation.")
+
+def write_calfits(
+    gains, 
+    filename, 
+    sim=None, 
+    freqs=None, 
+    times=None, 
+    clobber=False
+):
+    """Write gains to disk as a calfits file."""
+    from hera_cal.io import write_cal
+    if sim is not None:
+        if isinstance(sim, Simulator):
```

Comment:
I'm now slightly wary of passing around the `Simulator` class and modifying its `data` object, given our troubles in validation 4.0. Not sure it would be an issue here particularly, but we should keep an eye out for it.

Reply:
Agreed. It's not an issue here, since it's just used to pull required metadata if not provided by the other keywords, but the issues in the 4.0 test definitely warrant the wariness.